tests/Integration/OptionsTest.php

```diff
@@ -0,0 +1,175 @@
+<?php
+/**
+ * Integration Tests: Plugin options
+ *
+ * @package Parsely\Tests
+ */
+
+declare(strict_types=1);
+
+namespace Parsely\Tests\Integration;
+
+use Parsely\Parsely;
+
+/**
+ * Integration Tests for plugin options.
+ */
+final class OptionsTest extends TestCase {
+	/**
+	 * Internal variable.
+	 *
+	 * @var Parsely $parsely Holds the Parsely object.
+	 */
+	private static $parsely;
+
+	/**
+	 * Setup method called before each test.
+	 */
+	public function set_up(): void {
+		parent::set_up();
+
+		self::$parsely = new Parsely();
+	}
+
+	/**
+	 * Verifies that get_options() can handle corrupted or unset values.
+	 *
+	 * @since 3.0.0
+	 *
+	 * @covers \Parsely\Parsely::get_options
+	 */
+	public function test_default_options_are_returned_when_options_are_corrupted_or_not_set(): void {
+		update_option( Parsely::OPTIONS_KEY, 'someinvalidvalue' );
+
+		$options = self::$parsely->get_options();
+		self::assertSame( self::$parsely->get_default_options(), $options );
+
+		delete_option( Parsely::OPTIONS_KEY );
```

Comment:
This `delete_option` and the above `update_option` seems unnecessary because `get_options` function isn't setting any WP Option.

Reply:
I haven't performed any changes for the specific item, since I think that what is done here is correct. For everything else, I've applied changes that should address your feedback.